domain/src/main/java/com/open/domain/repository/WeatherRepository.kt

```diff
@@ -0,0 +1,10 @@
+package com.open.domain.repository
+
+
+import com.open.domain.Weather
+import dagger.Provides
+import kotlinx.coroutines.flow.Flow
+
+interface WeatherRepository {
+    fun getWeather(cityName: String): Flow<Weather>
```

Comment:
Just in case you're using `Flow` to catch potential errors, my recommendation would be to use kotlin's `Result` class instead.